### PR TITLE
fix variable shadowing

### PR DESCRIPTION
**Pull request by maxbachmann**

### CODE REVIEW

File: Cython/Utility/ExtensionTypes.c [hunk 1]
```diff
@@ -121,7 +121,7 @@ static int __Pyx_PyType_Ready(PyTypeObject *t) {
         (void)__Pyx_PyObject_CallMethod0;
 #endif
 
-    r = PyType_Ready(t);
+        r = PyType_Ready(t);
```
Comment (scoder) [inline Cython/Utility/ExtensionTypes.c:124]: This was probably intended (not just indented).
```suggestion
    r = PyType_Ready(t);
```

File: Cython/Utility/ExtensionTypes.c [hunk 2]
```diff
@@ -131,16 +131,16 @@ static int __Pyx_PyType_Ready(PyTypeObject *t) {
             PyGC_Enable();
     #else
         if (gc_was_enabled) {
-            PyObject *t, *v, *tb;
-            PyErr_Fetch(&t, &v, &tb);
+            PyObject *ty, *v, *tb;
+            PyErr_Fetch(&ty, &v, &tb);
```
Comment (scoder) [inline Cython/Utility/ExtensionTypes.c:135]: I think other places use `tp` for a type reference. Don't know of any that use `ty`.
```suggestion
            PyObject *tp, *v, *tb;
            PyErr_Fetch(&tp, &v, &tb);
```

File: Cython/Utility/ExtensionTypes.c [hunk 3]
```diff
@@ -131,16 +131,16 @@ static int __Pyx_PyType_Ready(PyTypeObject *t) {
...
             ret = __Pyx_PyObject_CallMethod0(gc, PYUNICODE("enable"));
             if (likely(ret || r == -1)) {
                 Py_XDECREF(ret);
                 // do not overwrite exceptions raised by PyType_Ready() above
-                PyErr_Restore(t, v, tb);
+                PyErr_Restore(ty, v, tb);
```
Comment (scoder) [inline Cython/Utility/ExtensionTypes.c:140]:
```suggestion
                PyErr_Restore(tp, v, tb);
```

File: Cython/Utility/ExtensionTypes.c [hunk 4]
```diff
@@ -131,16 +131,16 @@ static int __Pyx_PyType_Ready(PyTypeObject *t) {
...
             } else {
                 // PyType_Ready() succeeded, but gc.enable() failed.
-                Py_XDECREF(t);
+                Py_XDECREF(ty);
```
Comment (scoder) [inline Cython/Utility/ExtensionTypes.c:143]:
```suggestion
                Py_XDECREF(tp);
```

**Review by scoder — COMMENTED** [review]
Thanks for spotting this.

**Comment by maxbachmann** [comment]
> I think other places use tp for a type reference. Don't know of any that use ty.

Yes I simply did not find any other uses, so I selected the name without thinking much ;)

> This was probably intended (not just indented).

I added those spaces so it fits the overall indentation at this section of the functions

**Pull request merged**